src/Alert.php

```diff
@@ -0,0 +1,644 @@
+     * @param array $value
+     *
+     * @return static
+     *
+     * {@see Html::renderTagAttributes()} for details on how attributes are being rendered.
+     */
+    public function iconContainerAttributes(array $value): self
+    {
+        $new = clone $this;
+        $new->iconContainerAttributes = $value;
+        return $new;
+    }
+
+    /**
+     * The CSS class for the container `<i>` tag.
+     *
+     * @param string $value
+     *
+     * @return static
+     */
+    public function iconContainerClass(string $value): self
+    {
+        $new = clone $this;
+        $new->iconContainerClass = $value;
+        return $new;
+    }
+
+    /**
+     * Set icon text in the alert component.
+     *
+     * @param string $value
+     *
+     * @return static
+     */
+    public function iconText(string $value): self
+    {
+        $new = clone $this;
+        $new->iconText = $value;
+        return $new;
+    }
+
+    /**
+     * Set layout the alert panel body.
+     *
+     * @param string $value
+     *
+     * @return static
+     */
+    public function layoutBody(string $value): self
+    {
+        $new = clone $this;
+        $new->layoutBody = $value;
+        return $new;
+    }
+
+    /**
+     * Set layout the alert panel header.
+     *
+     * @param string $value
+     *
+     * @return static
+     */
+    public function layoutHeader(string $value): self
+    {
+        $new = clone $this;
+        $new->layoutHeader = $value;
+        return $new;
+    }
+
+    protected function run(): string
+    {
+        $new = clone $this;
+        return $new->renderAlert($new);
+    }
+
+    /**
+     * Render Alert.
+     */
+    private function renderAlert(self $new): string
+    {
+        $new->attributes['role'] = 'alert';
+        $new->attributes['id'] ??= $new->id ?? Html::generateId('alert-');
+
+        if (!isset($new->parts['{button}'])) {
+            $new->renderCloseButton($new);
+        }
+
+        if (!isset($new->parts['{icon}'])) {
+            $new->renderIcon($new);
+        }
+
+        if (!isset($new->parts['{body}'])) {
+            $new->renderBody($new);
+        }
+
+        if (!isset($new->parts['{header}'])) {
+            $new->renderHeader($new);
+        }
+
+        $contentAlert = $new->renderPanelHeader($new) . PHP_EOL . $new->renderPanelBody($new);
+
+        if ($new->class !== '') {
+            Html::addCssClass($new->attributes, $new->class);
+        }
+
+        return $new->body !== ''
+            ? Div::tag()
+                ->attributes($new->attributes)
+                ->content(PHP_EOL . trim($contentAlert) . PHP_EOL)
+                ->encode(false)
+                ->render()
+            : '';
+    }
+
+    /**
+     * Renders close button.
+     */
+    private function renderCloseButton(self $new): void
+    {
+        $new->parts['{button}'] = '';
+
+        if ($new->buttonOnClick !== '') {
+            $new->buttonAttributes['onclick'] = $new->buttonOnClick;
```

Comment:
Rendering should not modify attributes.

Reply:
https://github.com/terabytesoftw/yii-widgets/blob/7e1e5582d580b21b1c1b32afb8711612b950ea02/tests/AlertTest.php#L408 As you can see in the example for it is only configured if the user needs it, tailwind and bulma do not have JS and it is necessary, create the function so that it has ide support, and you do not have to pass it as an attribute.

Reply:
Fixed.